### ISSUE TITLE
Create the concept of a light

### ISSUE THREAD

**Issue by adampolar** [Steve58/aero-planar#1]
It would be good to have the concept of a point light. If planes are facing away they would become darker and if they were facing the point light they would become lighter.

**Comment by Steve58** [comment]
Good idea Adam. I think the functions in /engine/collision.js could be reworked to do the calculations:
- Set up polygon frame, check whether camera and light source are on the same or opposite sides of polygon.
- Tweak this code so the polygon frame origin is at the centre of the polygon.
- Then can work out angle and our distance from polygon to light, brightest when close and perpendicular?
- Other functions in collision.js to see if there are other polygons blocking the path from the light to the polygon.

Could even try angled gradients across single polygons when the angle is shallow? Might have performance issues?

Shaped shadows would be great but sounds trickier...
